azure/ConsiderationsForServiceDesign.md

```diff
@@ -141,94 +142,197 @@ cannot collide with a resource path that contains user-specified resource ids.
 Long-running operations are an API design pattern that should be used when the processing of
 an operation may take a significant amount of time -- longer than a client will want to block
 waiting for the result.
-Azure allows for two forms of this design pattern: resource-based long-running operations (RELO),
-which is the preferred pattern, and long-running operations with a status monitor.
+The request that initiates a long-running operation generally returns a `202 Accepted` response with a
+response body that contains a representation of a _status monitor_, which is an ephemeral resource that
+will track the status and final result of the operation.
+The status monitor resource is distinct from the target resource and specific to the individual
+operation request.
+
+This diagram illustrates how a long-running operation with a status monitor is initiated and then how the client
+determines it has completed and obtains its results:
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant API Endpoint
+    participant Status Monitor
+    Client->>API Endpoint: PUT/POST/DELETE
+    API Endpoint->>Client: HTTP/1.1 202 Accepted<br/>Retry-After: 5<br/>{ "id": "22", "status": "NotStarted" }
+    Client->>Status Monitor: GET
+    Status Monitor->>Client: HTTP/1.1 200 Ok<br/>Retry-After: 5<br/>{ "id": "22", "status": "Running" }
+    Client->>Status Monitor: GET
+    Status Monitor->>Client: HTTP/1.1 200 Ok<br/>Retry-After: 5<br/>{ "id": "22", "status": "Succeeded" }
+```
 
-In both patterns, the processing of the operation is initiated by one API call and the client
-obtains the results of the operation from a subsequent API call.
-Here we illustrate the sequence of API calls involved in each of these patterns.
+1. The client sends the request to initiate the long-running operation.
+The initial request could be a PUT, POST, or DELETE method.
+PATCH should not be used for LROs -- it should be reserved for simple resource updates.
+If a long-running update is required it should be implemented with POST.
+The request may contain an `operation-id` header that the service will use as the ID of the status monitor created for the operation.
 
-### Resource-based long-running operations
+2. The resource validates the request and initiates the operation processing.
+It sends a response to the client with a `202-Accepted` HTTP status code.
+The response body is status monitor for the operation including the ID, either from the request header or generated by the service.
+The response also includes a `Retry-after` header telling the client a minimum time to wait (in seconds)
+before sending a request to the status monitor URL.
+For backward compatibility, the response may also include an `Operation-Location` header containing the absolute URL
```

Comment:
I said "may" because Operation-Location is redundant with the information in the status monitor and thus not really necessary.  Our current SDK patterns use it so _for backward compatibility_ it is good to have but hopefully we'll update the generators soon so that they operate on the status monitor information and not the `Operation-Location`.

Reply:
I would strongly push for MUST include the Operation-Location. The operation location is not redundant in the response for anything that wants to act in a hyperlinked way. The status monitor resource by id only requires out-of-band knowledge in order to correctly form the URL. And while hyperlinking is not super commonly used in the general case, for single redirects etc. it has certainly proven to be a valuable concept in HTTP.